### DISCUSSION TITLE
Images。

### DISCUSSION THREAD

**Discussion by imageslr** [imageslr/blog#83] [Announcements]
https://imageslr.github.io/about.html 

 Software Elon Z

**Comment by TJKkking** [comment]
博主的博客好像不是部署在github page上，我很好奇你是把它部署到了自己的服务器上然后把xxx.github.io重定向到imageslr.com吗？我觉得蛮好玩的。请问你具体是怎么操作的呢？本地构建还是服务器上构建呀？发布文章的工作流是怎样的呢？以及涉及到了哪些工具和网站之类的？

**Comment by imageslr** [comment]
@TJKkking
1. 原来是部署在 github pages + netlify，不需要任何配置，就能自动部署。直接使用 github.io 的域名。全免费。
2. 后来为了使用自定义域名、同时提高国内访问速度，就部署到自己的服务器上了。这里把 github.io 域名下的所有静态文件，全部加了个 302 重定向 js 脚本，自动跳转到新域名。
3. 新域名的自动部署主要是基于 Jenkins + Gitee Hooks。简单来说：
   1. 先配置 jenkins，执行 jekyll 的编译命令就好；
   2. 配置 github 和 gitee 自动单向同步（国内服务器无法访问 github）；
   3. gitee 配置一个 actions，收到 push 事件时触发 jenkins 脚本，完成部署。
   4. 申请一个飞书机器人 token，在部署完成时，通过 curl 发送一个消息通知。

可以参考这篇文章：https://itiandong.com/2021/automating-blog-deployment-with-jenkins-and-gitee/

**Reply by imageslr** [reply]
发布了新的文章：[使用 Netlify + Github 快速搭建个人博客](https://imageslr.com/2023/jekyll-netlify.html)

**Comment by JakATom** [comment]
博主的主题好喜欢，求分享，谢谢

**Comment by imageslr** [comment]
@JakATom   https://github.com/huangyz0918/moving

**Comment by Guochengjie** [comment]
博主您好，我很喜欢您的写的Indigo主题Markdown预览CSS，但是您没写开源协议。可否允许在标注原作者的前提下将您的CSS代码迁移到https://github.com/PrivateBin/PrivateBin这个项目中呢？

**Reply by imageslr** [reply]
请随意使用
